### ISSUE TITLE
Few questions about difference from original implementation

### ISSUE THREAD

**Issue by jackaceuser** [sunset1995/pytorch-layoutnet#2]
Hi

Thank you very much for sharing
I've read through your code (and was very impressed!) and collected several questions about difference from the original implementation.
I'm wondering whether I'm right and the differences exist, and why you have chosen to implement it this way

1. The learning rate that you use is very different. You use ~0.018 (after warmup), while the original work uses 0.0001 (times 5, because they multiply by 5 where the mask is true, while you divide by 5 where it is false). That is a very significant factor of 36... am I missing something? Maybe something is different between torch in lua and pytorch?

2. The original work first trains only on edges, and then trains both on edges and on corners. Why have you chosen to avoid this?

3. You haven't implemented the gamma augmentation (But I'm quite sure that it is not very significant)

4. You use weight decay. Why?

5. You use learning rate decay. Why?

6. The original work clamped the gradients at 10. You have clipped the gradients' norm (Not the individual gradients) at 3. Any reason you have chosen to change this value?

The questions obviously don't criticize your work, just trying to understand the differences and get the best of both worlds...

Thanks

**Comment by sunset1995** [comment]
Hello,
Thank you very much for your appreciation.
1) It follows [original paper](http://openaccess.thecvf.com/content_cvpr_2018/papers/Zou_LayoutNet_Reconstructing_the_CVPR_2018_paper.pdf) sec 3.3 which says the learning rate is set to "e^-4". I interpret "e" as Euler's number. (so it that actually "10"?) Two reasons why it still work are (1) Adam (original) vs SGD (this) and default `lr` for Adam in Pytorch is lower than others. (2) The learning rate schedule (warmup + poly decay) help convergence.
2) I prefer train a network in one shot. My first trial to attack this is learning rate schedule + SGD with momentum and found it work.
3) I just add this in commit [0f7b879](https://github.com/sunset1995/pytorch-layoutnet/commit/0f7b87970903f2e989fe511fb2a41acf35ed525d)
4) I expect this components could get better result according to my experience. At least, my first trial with them work
5) same as 4)
6) I didn't note that. Their paper doesn't mention this so I add this component intuitively.

Hope my answer help and have a nice day :)

**Comment by jackaceuser** [comment]
About 1 - Yes, they use 0.0001. From their implementation in file (model_pano_joint.lua):

    cmd:option('-lr' , 1e-4, 'learning rate')

Still seems very different from the number you use. My guess is that there is some difference between torch in lua and pytorch, but I have no idea. Or maybe it's related to the difference between Adam and SGD.
